pkg/utils/utils.go

```diff
@@ -19,3 +28,125 @@ func GetOCMAccessToken() (*string, error) {
+		err = fmt.Errorf("Can't retrieve subscription for key '%s': %v", key, err)
+		return
+	}
+
+	// If there is exactly one matching subscription then return the corresponding cluster:
+	subsTotal := subsListResponse.Total()
+	if subsTotal == 1 {
+		id, ok := subsListResponse.Items().Slice()[0].GetClusterID()
+		if ok {
+			var clusterGetResponse *cmv1.ClusterGetResponse
+			clusterGetResponse, err = clustersResource.Cluster(id).Get().
+				Send()
+			if err != nil {
+				err = fmt.Errorf(
+					"Can't retrieve cluster for key '%s': %v",
+					key, err,
+				)
+				return
+			}
+			cluster = clusterGetResponse.Body()
+			return
+		}
+	}
+
+	// If there are multiple subscriptions that match the cluster then we should report it as
+	// an error:
+	if subsTotal > 1 {
+		err = fmt.Errorf(
+			"There are %d subscriptions with cluster identifier or name '%s'",
+			subsTotal, key,
+		)
+		return
+	}
+
+	// If we are here then no subscription matches the passed key. It may still be possible that
+	// the cluster exists but it is not reporting metrics, so it will not have the external
+	// identifier in the accounts management service. To find those clusters we need to check
+	// directly in the clusters management service.
+	clustersSearch := fmt.Sprintf(
+		"id = '%s' or name = '%s' or external_id = '%s'",
+		key, key, key,
+	)
+	clustersListResponse, err := clustersResource.List().
+		Search(clustersSearch).
+		Size(1).
+		Send()
+	if err != nil {
+		err = fmt.Errorf("Can't retrieve clusters for key '%s': %v", key, err)
+		return
+	}
+
+	// If there is exactly one cluster matching then return it:
+	clustersTotal := clustersListResponse.Total()
+	if clustersTotal == 1 {
+		cluster = clustersListResponse.Items().Slice()[0]
+		return
+	}
+
+	// If there are multiple matching clusters then we should report it as an error:
+	if clustersTotal > 1 {
+		err = fmt.Errorf(
+			"There are %d clusters with identifier or name '%s'",
+			clustersTotal, key,
+		)
+		return
+	}
+
+	// If we are here then there are no subscriptions or clusters matching the passed key:
+	err = fmt.Errorf(
+		"There are no subscriptions or clusters with identifier or name '%s'",
+		key,
+	)
+	return
+}
+
+func GetClusterLimitedSupportReasons(connection *sdk.Connection, clusterID string) ([]*lmtSprReasonItem, error) {
+
+	limitedSupportReasons, err := connection.ClustersMgmt().V1().
+		Clusters().
+		Cluster(clusterID).
+		LimitedSupportReasons().
+		List().
+		Send()
+	if err != nil {
+		return nil, fmt.Errorf("Failed to get limited Support Reasons: %s", err)
+	}
+
+	lmtReason := limitedSupportReasons.Items()
+
+	var clusterLmtSprReasons []*lmtSprReasonItem
+
+	lmtReason.Each(func(lmtSprReason *cmv1.LimitedSupportReason) bool {
+		clusterLmtSprReason := lmtSprReasonItem{
+			ID:      lmtSprReason.ID(),
+			Summary: lmtSprReason.Summary(),
+			Details: lmtSprReason.Details(),
+		}
+		clusterLmtSprReasons = append(clusterLmtSprReasons, &clusterLmtSprReason)
+
+		return true
+	})
```

Comment:
```suggestion
	for _, resp := range lmtReason {
		clusterLmtSprReason := lmtSprReasonItem{
			ID:      resp.ID(),
			Summary: resp.Summary(),
			Details: resp.Details(),
		}
		clusterLmtSprReasons = append(clusterLmtSprReasons, &clusterLmtSprReason)
	}
```

I think putting this directly in a loop is better than using an anonymous function if we're only ever going to return true

Reply:
I tried like this, this will not work, the range function is trying then to iterate through a pointer which is what the lmtReason is.

Reply:
Looks like you can use `UnmarshalLimitedSupportReasonList` to get a list of limited support reasons, and then iterate through that. 

 https://github.com/openshift-online/ocm-sdk-go/blob/main/clustersmgmt/v1/limited_support_reason_list_type_json.go#L56